packages/go_router/lib/src/configuration.dart

```diff
@@ -0,0 +1,149 @@
+  }) {
+    _cacheNameToPath('', routes);
+    assert(() {
+      _debugKnownRoutes();
+      return true;
+    }());
+  }
+
+  /// List of top level routes used by the go router delegate
+  final List<GoRoute> routes;
+
+  /// The limit for the number of consecutive redirects
+  final int redirectLimit;
+
+  /// Top level page redirect.
+  final GoRouterRedirect topRedirect;
+
+  final Map<String, String> _nameToPath = <String, String>{};
+
+  /// Throws a [RouteConfigurationError] if this configuration is invalid
+  void validate() {
+    for (final GoRoute route in routes) {
+      if (!route.path.startsWith('/')) {
+        throw RouteConfigurationError(
+            'top-level path must start with "/": ${route.path}');
+      }
+    }
+  }
+
+  /// Looks up the url location by a [GoRoute]'s name
+  String namedLocation(
+    String name, {
+    Map<String, String> params = const <String, String>{},
+    Map<String, String> queryParams = const <String, String>{},
+  }) {
+    assert(() {
+      log.info('getting location for name: '
+          '"$name"'
+          '${params.isEmpty ? '' : ', params: $params'}'
+          '${queryParams.isEmpty ? '' : ', queryParams: $queryParams'}');
+      return true;
+    }());
+    assert(_nameToPath.containsKey(name), 'unknown route name: $name');
+    final String path = _nameToPath[name]!;
+    assert(() {
+      // Check that all required params are present
+      final List<String> paramNames = <String>[];
+      patternToRegExp(path, paramNames);
+      for (final String paramName in paramNames) {
+        assert(params.containsKey(paramName),
+            'missing param "$paramName" for $path');
+      }
+
+      // Check that there are no extra params
+      for (final String key in params.keys) {
+        assert(paramNames.contains(key), 'unknown param "$key" for $path');
+      }
+      return true;
+    }());
+    final Map<String, String> encodedParams = <String, String>{
+      for (final MapEntry<String, String> param in params.entries)
+        param.key: Uri.encodeComponent(param.value)
+    };
+    final String location = patternToPath(path, encodedParams);
+    return Uri(path: location, queryParameters: queryParams).toString();
+  }
+
+  @override
+  String toString() {
+    return 'RouterConfiguration: $routes';
+  }
+
+  String _debugKnownRoutes() {
+    final StringBuffer sb = StringBuffer();
+    sb.writeln('Full paths for routes:');
+    _debugFullPathsFor(routes, '', 0, sb);
+
+    if (_nameToPath.isNotEmpty) {
+      sb.writeln('known full paths for route names:');
+      for (final MapEntry<String, String> e in _nameToPath.entries) {
+        sb.writeln('  ${e.key} => ${e.value}');
+      }
+    }
+
+    return sb.toString();
+  }
+
+  void _debugFullPathsFor(
+      List<GoRoute> routes, String parentFullpath, int depth, StringBuffer sb) {
+    for (final GoRoute route in routes) {
+      final String fullpath = concatenatePaths(parentFullpath, route.path);
+      assert(() {
```

Comment:
Is this `assert` intentional now that this is not logging directly?

Reply:
I'm not sure why this named location feature needs so many `asserts`.  I'd like to simplify how `namedLocation` works or remove the API altogether.